test/k4FWCoreTest/CMakeLists.txt

```
@@ -221,3 +221,11 @@ add_test_with_env(TypeMisMatchDemo options/TypeMisMatchDemo.py PROPERTIES PASS_R
 
 add_test_with_env(TypeMisMatchDemoMultiple options/TypeMisMatchDemoMultiple.py PROPERTIES PASS_REGULAR_EXPRESSION "Failed to cast collection MCParticles1 to the required type")
 add_test_with_env(ReadMarlinWrapperCollection options/readMarlinWrapperCollection.py)
+
+add_test(NAME check_broken_pipe
+         COMMAND bash -c "${K4RUN} options/ExampleFunctionalProducer.py | head -n 2 | wc -l"
```

```suggestion
         COMMAND bash -c "[ $(${K4RUN} options/ExampleFunctionalProducer.py | head -n 2 | wc -l) = 2 ]"
```
Would make the it possible to go without the regular expression checking below